<!--aribo-gmappingLaunch.launch-->
<launch>

<!-- declare arguments with default values -->
<arg name="pcap" default="" />
<arg name="calibration" default="$(find velodyne_pointcloud)/params/32db.yaml"/>
<arg name="min_range" default="0.4" />
<arg name="max_range" default="130.0" />

  <!-- start nodelet manager and driver nodelets -->
  <include file="$(find velodyne_driver)/launch/nodelet_manager.launch">
    <arg name="model" value="32E"/>
    <arg name="pcap" value="$(arg pcap)"/>
  </include>

  <!-- start cloud nodelet -->
  <include file="$(find velodyne_pointcloud)/launch/cloud_nodelet.launch">
    <arg name="calibration" value="$(arg calibration)"/>
    <arg name="min_range" value="$(arg min_range)"/>
    <arg name="max_range" value="$(arg max_range)"/>
  </include>

  <!-- convert the 3d pointcloud into 2d laser scan -->
    <node name="pointcloud_to_laserscan" pkg="pointcloud_to_laserscan" type="pointcloud_to_laserscan_node">

        <remap from="cloud_in" to="/velodyne_points"/>
        <param name="output_frame_id" value="/velodyne_laser_static"/>    
 <!--   <param name="min_height" value="-1"/>
    <param name="max_height" value="3"/> -->
        <param name="angle_min" value="-3.14159"/>
        <param name="angle_max" value="3.14159"/>
        <param name="scan_time" value="1/1000.0"/>
        <param name="range_min" value="0"/>
        <param name="range_max" value="50"/>

    </node>

  <node type="rviz" name="rviz" pkg="rviz" args="-d $(find usma_velodyne)/launch/velodyne_bare.rviz" />

</launch>
